.github/workflows/build.yml

```diff
@@ -39,6 +39,24 @@ jobs:
         run: make build-linux-amd64
       - name: Run Unit Tests
         run: make test
+      - name: Run E2E Tests
+        run: |
+          cd $GITHUB_WORKSPACE/test/e2e
+          go install github.com/onsi/ginkgo/v2/ginkgo
+          tar -xvzf testdata/distribution/docker.tar.gz -C $GITHUB_WORKSPACE/test/e2e/testdata/distribution/
+          trap 'docker kill oras-e2e || true' ERR
+          docker run -d -p 5000:5000 --rm --name oras-e2e \
+            --env STORAGE_DELETE_ENABLED=true \
+            --env AUTH_HTPASSWD_REALM=test-basic \
+            --env AUTH_HTPASSWD_PATH=/etc/docker/registry/passwd \
```

Comment:
Let's remove `passwd` in this PR and move it to other RPs like testing `oras login`.